src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.Api/Controllers/AuthorizationController.cs

```diff
@@ -51,11 +51,11 @@ public async Task<ActionResult> GetCurrentParty(bool returnPartyObject = false)
             UserContext userContext = await _userHelper.GetUserContext(HttpContext);
             int userId = userContext.UserId;
             string cookieValue = Request.Cookies[_settings.GetAltinnPartyCookieName];
-            int.TryParse(cookieValue, out int partyId);
+            int.TryParse(cookieValue, out int partyIdFromCookie);
 
-            if (partyId != 0)
+            if (userContext.UserParty == null || userContext.PartyId != userContext.UserParty.PartyId)
```

Comment:
Maybe a comment like this? 
```suggestion
            // Currently selected party is different from authenticated user.
            if (userContext.UserParty == null || userContext.PartyId != userContext.UserParty.PartyId)
```
It is hard to understand what this if actually tests. Party, party, party. :)

Reply:
Added more text